test/testInnosentStateMachine.js

```diff
@@ -0,0 +1,275 @@
+const chai = require('chai')
+const { describe, it } = require('mocha')
+const moment = require('moment')
+
+const expect = chai.expect
+const STATE = require('../SessionStateEnum.js')
+const OD_FLAG_STATE = require('../SessionStateODFlagEnum.js')
+const DOOR_STATE = require('../SessionStateDoorEnum.js')
+const StateMachine = require('../InnosentStateMachine.js')
+
+function setupDB(location_data = {}, session = {}, is_overdose_suspected = false) {
+  return {
+    getLocationData() {
+      return location_data
+    },
+    getMostRecentSession() {
+      return session
+    },
+    isOverdoseSuspectedInnosent() {
+      return is_overdose_suspected
+    },
+  }
+}
+
+function setupRedis(states = {}, door = {}, innosentHistory = [0], innosent = {}) {
+  return {
+    getLatestLocationStatesData() {
+      // initial state
+      return states
+    },
+    getLatestInnosentSensorData() {
+      return innosent
+    },
+    getLatestDoorSensorData() {
+      return door
+    },
+    addStateMachineData() {},
+    getInnosentWindow() {
+      return innosentHistory
+    },
+  }
+}
+
+describe('test getNextState', () => {
```

Comment:
Since this is copied from `testXeThruStateMachine.js`, it has the same top level name. I see that you've specified innosent in a few of the lower-level test names, but I think it should be done here too otherwise it might not be clear which `test getNextState` --> `when no initial state` --> `should return the RESET state` test was run, for example.